Documentation/botr/readytorun-format.md

```diff
@@ -460,7 +460,50 @@ the first byte of the encoding specify the number of following bytes as follows:
 
 ## Hashtable
 
-**TODO**: Document native format hashtable
+Conceptually, a native hash table is a header that describe the dimensions of the table, a table that maps hash values of the keys to buckets followed with a list of buckets that store the values. These three things are stored consecutively in the format.
+
+To make look up fast, the number of buckets is always a power of 2. The table is simply a sequence of `(1 + number of buckets)` cells, for the first `(number of buckets)` cells, its stores the offset of the bucket list from the beginning of the whole native hash table. The last cell stores the offset to the end of the buckets.
+
+Each bucket is a sequence of entries. An entry has a hash code and an offset to the object stored. The entries are sorted by hash code.
+
+Physically, the header is a single byte. The most significant six bits is used to store the number of buckets in its base-2 logarithm. The remaining two bits are used for storing the entry size, as explained below:
+
+Because the offsets to the bucket lists are often small numbers, the table cells are variable sized. 
+It could be either 1 byte, 2 bytes or 4 bytes. The three cases are described with two bits. `00` means it is one byte, `01` means it is two bytes and `02` means it is four bytes.
```

Comment:
`02`?

Reply:
Thanks for the catch, of course, I meant `10`